client/src/driver/BitBakeProjectScanner.ts

```diff
@@ -412,41 +412,81 @@ You should adjust your docker volumes to use the same URIs as those present on y
   }
 
   private async scanForRecipesPath (): Promise<void> {
-    const tmpFiles = this.searchFiles(this._recipesFileExtension)
+    const showRecipeFileNameOutput = await this.executeBitBakeCommand('bitbake-layers show-recipes -f')
+    if (showRecipeFileNameOutput.status !== 0) {
+      logger.error('Failed to scan recipes path')
+      return
+    }
 
-    for (const file of tmpFiles) {
-      const recipeName: string = file.name.split(/[_]/g)[0]
+    const output = showRecipeFileNameOutput.output.toString()
+    const splittedOutput = output.split(/\r?\n/g)
+    const startingIndex = splittedOutput.findIndex((line) => line.includes('Available recipes'))
 
-      const element: ElementInfo | undefined = this._bitbakeScanResult._recipes.find((obj: ElementInfo): boolean => {
-        return obj.name === recipeName
-      })
+    const allFileNames: string[] = []
+    if (startingIndex === -1) {
+      logger.error('[scanForRecipesPath] Failed to find available recipes')
+      return
+    }
+
+    const recipePathRegex = /(.*\.bb)/
 
-      if (element !== undefined) {
-        element.path = file.path
+    // All lines after the '=== Available recipes: ===' line are file names, and only keep the valid ones
+    allFileNames.push(...splittedOutput.slice(startingIndex + 1).filter((line) => !line.includes('skipped') && recipePathRegex.exec(line) !== null))
```

Comment:
You could be more selective and use "(skipped:".
This would allow tricky recipes like libskipped.bb to still get registered

Reply:
Good catch, done.